mlflow/pipelines/steps/train.py

```diff
@@ -383,6 +365,58 @@ def _run(self, output_directory):
 
         return card
 
+    def _get_user_defined_estimator(self, X_train, y_train, validation_df, run, output_directory):
+        train_module_name, estimator_method_name = self.step_config["estimator_method"].rsplit(
+            ".", 1
+        )
+        sys.path.append(self.pipeline_root)
+        estimator_fn = getattr(importlib.import_module(train_module_name), estimator_method_name)
+        estimator_hardcoded_params = self.step_config["estimator_params"]
+        if self.step_config["tuning_enabled"]:
+            best_estimator_params = self._tune_and_get_best_estimator_params(
+                run.info.run_id,
+                estimator_hardcoded_params,
+                estimator_fn,
+                X_train,
+                y_train,
+                validation_df,
+                output_directory,
+            )
+            estimator = estimator_fn(best_estimator_params)
+        elif len(estimator_hardcoded_params) > 0:
+            estimator = estimator_fn(estimator_hardcoded_params)
+        else:
+            estimator = estimator_fn()
+        return estimator
+
+    def _resolve_estimator_plugin(self, plugin_str, X_train, y_train):
+        plugin_str = plugin_str.replace("/", ".").replace("@", ".")
```

Comment:
Why do we need this? Can we just make the user plugin str `automl.flaml`?

Reply:
The idea is to have some sort of a plugin mechanism so that in the future, if we decide to add AutoGluon, for example, we can just implent autogluon.py, and "automl/autogluon" will work out of the box.